src/tray.ts

```diff
@@ -12,112 +12,68 @@ const getIcon = (): string => {
-      click: () => {
-        run('startTelework');
-      },
-    },
-    {
-      type: 'normal',
-      label: '退勤打刻',
-      click: () => {
-        run('finishWork');
-      },
-    },
-    {
-      type: 'separator',
-    }
-  );
-
-  if (showDirectly) {
-    menu.push(
-      {
-        type: 'normal',
-        label: '直行打刻',
-        click: () => {
-          run('startWorkDirectly');
-        },
-      },
-      {
-        type: 'normal',
-        label: '直帰打刻',
-        click: () => {
-          run('finishWorkDirectly');
-        },
-      },
-      {
-        type: 'separator',
-      }
-    );
-  }
-
-  menu.push(
-    {
-      type: 'normal',
-      label: '私用外出開始',
-      click: () => {
-        run('pauseWork');
-      },
-    },
-    {
-      type: 'normal',
-      label: '私用外出終了',
-      click: () => {
-        run('restartWork');
-      },
-    },
-    {
-      type: 'separator',
-    },
-    {
-      type: 'normal',
-      label: 'ログイン・設定',
-      click: open,
-    },
-    {
-      type: 'separator',
-    },
-    {
-      type: 'normal',
-      label: 'AKASHI',
-      click: () => {
-        shell.openExternal('https://atnd.ak4.jp/login');
-      },
-    }
-  );
-  if (release) {
-    menu.push({
-      type: 'normal',
-      label: `🌟DakokuApp: ${release.name}を入手する🌟`,
-      click: () => {
-        shell.openExternal(release.html_url);
-      },
-    });
-  }
-  menu.push({
-    type: 'normal',
-    label: '終了',
-    click: app.quit,
+  const separator = generateMenuItem('separator');
+  const startWork = generateMenuItem('normal', '出勤打刻', () => {
+    run('startWork');
   });
-
+  const finishWork = generateMenuItem('normal', '退勤打刻', () => {
+    run('finishWork');
+  });
+  const startWorkDirectly = generateMenuItem('normal', '直行打刻', () => {
+    run('startWorkDirectly');
+  });
+  const finishWorkDirectly = generateMenuItem('normal', '直帰打刻', () => {
+    run('finishWorkDirectly');
+  });
+  const pauseWork = generateMenuItem('normal', '私用外出開始', () => {
+    run('pauseWork');
+  });
+  const restartWork = generateMenuItem('normal', '私用外出終了', () => {
+    run('restartWork');
+  });
+  const loginAndSetting = generateMenuItem('normal', 'ログイン・設定', open);
+  const akashi = generateMenuItem('normal', 'AKASHI', () => {
+    shell.openExternal('https://atnd.ak4.jp/login');
+  });
+  const release = generateMenuItem('normal', `🌟DakokuApp: ${storeRelease?.name}を入手する🌟`, () => {
+    shell.openExternal(storeRelease?.html_url + '');
+  });
+  const quit = generateMenuItem('normal', '終了', app.quit);
+  const itemAndCondition: [MenuItemConstructorOptions, boolean][] = [
+    [startWork, true],
+    [finishWork, true],
+    [separator, true],
+    [startWorkDirectly, showDirectly],
+    [finishWorkDirectly, showDirectly],
+    [separator, showDirectly],
+    [pauseWork, true],
+    [restartWork, true],
+    [separator, true],
+    [loginAndSetting, true],
+    [separator, true],
+    [akashi, true],
+    [release, !!storeRelease],
```

Comment:
```suggestion
    [release, storeRelease !== null],
```

間違いではないのですが、ちゃんと比較したさがあります